app/controllers/offers_controller.rb

```diff
@@ -0,0 +1,26 @@
+class OffersController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @suggested_lot = Lot.find(offer_params[:suggested_lot_id])
+    if @suggested_lot.present? && @suggested_lot.user == current_user
```

Comment:
`@suggested_lot` always presents at this line. If there is no such lot then `Lot.find()` will raise NotFound error